shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -1272,65 +1284,220 @@ - (void)updateViewportPadding {
-  id isLocal = info[UIKeyboardIsLocalUserInfoKey];
-  if (isLocal && ![isLocal boolValue]) {
-    return;
-  }
+- (void)keyboardWillBeHidden:(NSNotification*)notification {
+  // When keyboard is hidden or undocked, this notification will be triggered.
+  [self handleKeyboardNotification:notification notificationName:UIKeyboardWillHideNotification];
+}
 
-  // Ignore keyboard notifications if engine’s viewController is not current viewController.
-  if ([_engine.get() viewController] != self) {
+- (void)handleKeyboardNotification:(NSNotification*)notification
+                  notificationName:(NSNotificationName)notificationName {
+  BOOL ignoreNotification = [self shouldIgnoreKeyboardNotification:notification
+                                                  notificationName:notificationName];
+  if (ignoreNotification) {
     return;
   }
 
+  NSDictionary* info = notification.userInfo;
   CGRect keyboardFrame = [[info objectForKey:UIKeyboardFrameEndUserInfoKey] CGRectValue];
-  CGRect screenRect = [[UIScreen mainScreen] bounds];
+  FlutterKeyboardMode keyboardMode = [self calculateKeyboardAttachMode:keyboardFrame
+                                                      notificationName:notificationName];
+  CGFloat calculatedInset = [self calculateKeyboardInset:keyboardFrame keyboardMode:keyboardMode];
 
-  // Get the animation duration
+  // Avoid double triggering startKeyBoardAnimation.
+  if (self.targetViewInsetBottom == calculatedInset) {
+    return;
+  }
+
+  self.targetViewInsetBottom = calculatedInset;
   NSTimeInterval duration =
       [[info objectForKey:UIKeyboardAnimationDurationUserInfoKey] doubleValue];
-
-  // Considering the iPad's split keyboard, Flutter needs to check if the keyboard frame is present
-  // in the screen to see if the keyboard is visible.
-  if (CGRectIntersectsRect(keyboardFrame, screenRect)) {
-    CGFloat bottom = CGRectGetHeight(keyboardFrame);
-    CGFloat scale = [UIScreen mainScreen].scale;
-    // The keyboard is treated as an inset since we want to effectively reduce the window size by
-    // the keyboard height. The Dart side will compute a value accounting for the keyboard-consuming
-    // bottom padding.
-    self.targetViewInsetBottom = bottom * scale;
-  } else {
-    self.targetViewInsetBottom = 0;
-  }
   [self startKeyBoardAnimation:duration];
 }
 
-- (void)keyboardWillBeHidden:(NSNotification*)notification {
-  NSDictionary* info = [notification userInfo];
+- (BOOL)shouldIgnoreKeyboardNotification:(NSNotification*)notification
+                        notificationName:(NSNotificationName)notificationName {
+  BOOL isKeyboardNotificationForThisView = [self isKeyboardNotificationForThisView:notification];
+  BOOL isKeyboardRotated = [self isKeyboardRotated:notification];
+
+  // Don't ignore UIKeyboardWillHideNotification notifications.
+  if (notificationName == UIKeyboardWillHideNotification) {
+    // Skip hide notification when rotation in progress unless triggered by another app.
```

Comment:
Is this comment ambiguous? (Or maybe it is just me) I read "the rotation is triggered by another app" but I think you meant the hide keyboard notification is triggered by another app?